tests/test_web/test_webapi.py

```diff
@@ -232,8 +278,8 @@ def mock_webapi(
 def test_source_validation(mock_upload):
     sim = make_sim().copy(update={"sources": []})
     assert upload(sim, TASK_NAME, PROJECT_NAME, source_required=False)
-    with pytest.raises(SetupError):
-        upload(sim, TASK_NAME, PROJECT_NAME)
+    # with pytest.raises(SetupError):
+    # upload(Tidy3dStub(component=sim), TASK_NAME, PROJECT_NAME)
```

Comment:
Isn't it a problem if the user uploads a simulation without a source?

Reply:
yea this should probably not be commented out. I think this is caught as a `pre_upload_validator()`, which we should still be doing. Let's add this test back @magiWei

Reply:
Modified.